bazel_tools/client_server_test/runner/Main.hs

```diff
@@ -42,33 +37,31 @@ readPortFile n file =
       readPortFile (n-1) file
     Just p -> pure p
 
-runCheckedProc :: T.Text -> CreateProcess -> IO () -> IO ()
-runCheckedProc desc proc resume =
-  withCheckedProcessCleanup proc $ \(_ :: Handle) stdoutSrc stderrSrc ->
-  withAsync (runConduit (stderrSrc .| splitOutput .| C.awaitForever printStderr)) $ \_ ->
-  withAsync (runConduit (stdoutSrc .| splitOutput .| C.awaitForever printStdout)) $
-  const resume
-
-  where
-    printStdout line =
-      liftIO $ do
-        T.hPutStrLn stdout $ desc <> ": " <> line
-        hFlush stdout
-    printStderr line =
-      liftIO $ T.hPutStrLn stderr $ desc <> ": " <> line
-    splitOutput = C.T.decode C.T.utf8 .| C.T.lines
-
 main :: IO ()
 main = do
   [clientExe, clientArgs, serverExe, serverArgs] <- getArgs
   tempFile <- emptySystemTempFile "client-server-test-runner-port"
   let splitArgs = filter (/= "") . splitOn " "
 
-  let serverProc =
-        proc serverExe (["--port-file", tempFile] <> splitArgs serverArgs)
-  runCheckedProc "SERVER" serverProc $ do
+  -- `create_group` is needed on Windows to pass CTRL_BREAK_EVENT to the server process using `interruptProcessGroupOf`
+  let serverProc = (proc serverExe (["--port-file", tempFile] <> splitArgs serverArgs)) { create_group = True }
+  withProcess serverProc $ \_ -> do
     port <- readPortFile maxRetries tempFile
     removeFile tempFile
-    callProcess clientExe
-      (["--target-port", show port] <> splitArgs clientArgs)
-    exitSuccess
+    --  If the command run by `callProcess` returns a non-zero exit code, an exception is raised.
+    callProcess clientExe (["--target-port", show port] <> splitArgs clientArgs)
+    return ()
+
+withProcess :: CreateProcess -> ((Maybe Handle, Maybe Handle, Maybe Handle, ProcessHandle) -> IO a) -> IO a
+withProcess cp f =
+    bracket (createProcess cp)
```

Comment:
Are you reinventing withCreateProcess?

Reply:
It seems so. I'll double check if it will terminate the JVM without an interrupt sent in the case of an exception in the callback and adapt accordingly.